src/commons_abstract_fevalues.jl

```diff
@@ -186,21 +177,19 @@ where ``\\mathbf{u}_i`` are the nodal values of ``\\mathbf{u}``.
 @inline function function_gradient{dim, T}(fe_v::AbstractFEValues{dim}, q_point::Int, u::Vector{T})
     n_base_funcs = getnbasefunctions(getfunctionspace(fe_v))
     @assert length(u) == n_base_funcs
-    dN = shape_gradient(fe_v, q_point)
     grad = zero(Vec{dim, T})
     @inbounds for i in 1:n_base_funcs
-        grad += dN[i] * u[i]
+        grad += shape_gradient(fe_v, q_point, i) * u[i]
     end
     return grad
 end
 
 @inline function function_gradient{dim, T}(fe_v::AbstractFEValues{dim}, q_point::Int, u::Vector{Vec{dim, T}})
     n_base_funcs = getnbasefunctions(getfunctionspace(fe_v))
     @assert length(u) == n_base_funcs
-    dN = shape_gradient(fe_v, q_point)
     grad = zero(Tensor{2, dim, T})
-    for i in 1:n_base_funcs
-        grad += u[i] ⊗ dN[i]
+    @inbounds for i in 1:n_base_funcs
+        grad += u[i] ⊗ shape_gradient(fe_v, q_point, i)
```

Comment:
This method won't work I think. Atleast the \otimes should be a regular *
the input `u` should also now be a regular vector of values, not a vector of `Vec`'s.
We can probably just combined the two `function_gradient` methods now.